Update History for 2.0 release

tpike3 opened the pull request:
- updated history per #1707 
- included #1726 assuming integration of adv_tutorial (would wait to merge until #1726 merged)

rht commented on HISTORY.rst:
```diff
@@ -3,6 +3,60 @@
+
+**Breaking Changes:**
+
+* space: change `coord_iter` to return `(content,(x,y))` instead of `(content, [x],[y])`; this reduces known errors of scheduler to grid mismatch #1566, #1723
```
Should be `(content, x, y)` for the former version.

rht commented on HISTORY.rst:
```diff
@@ -3,6 +3,60 @@
+* space: change `coord_iter` to return `(content,(x,y))` instead of `(content, [x],[y])`; this reduces known errors of scheduler to grid mismatch #1566, #1723
+* space: change NetworkGrid `get_neighbors` to `get_neighborhood`; improves performance #1542
+* space: raise exception when pos is out of bounds in `Grid.get_neighborhood` #1524
+* space: remove deprecations (`find_empty` -> `move_to_empty`, num_agents -> removed param from `move_to_empty`, `position_agent` -> `place_agent`, `neighbor_iter` -> `iter_neighborhood`); complete deprecated function removal #1520, #1687, #1688
```
I find this one too cryptic.

tpike3 replied:
Line 18? what if it reads as 

 space: remove deprecations (use `move_to_empty` instead of `find_empty`, use  `position_agent`  instead of `place_agent` use `iter_neighborhood` instead of `neighbor_iter`;  `move_to_empty` no longer has the parameter num_agents (int)); complete deprecated function removal #1520, #1687, #1688

rht replied:
You could format them as a list (nested). This should be considered as a recipe for migration. As such, you should say the obsolete function first, then the replacer, because people are going to follow it step by step.

tpike3 replied:
So... 

space: remove deprecations [(`find_empty()` convert to `move_to_empty()`), (remove `num_agents` as a parameter from `move_to_empty()`), (`position_agent()` convert to  `place_agent()`), (`neighbor_iter()` convert to `iter_neighborhood()`); complete deprecated function removal #1520, #1687, #1688

rht replied:
Uh, I meant for it to be read by humans, not LLMs.
I meant list in a list:
* Remove deprecated functions.
  - `find_empty`: convert this to `move_to_empty`
  - `position_agent`: convert this to `place_agent`

I think it is plausible that the LLMs can automate the migration of existing code, if they are fed the old API and new API data, and the migration guide.

rht commented on HISTORY.rst:
```diff
@@ -3,6 +3,60 @@
+    * improve regex for better build #1669, #1671
+    * exclude notebooks form linter #1670
+    * updated pip for zsh #1644
+    * CLI quality of life improvements #1640
```
Is this CI or CLI?

rht commented on HISTORY.rst:
```diff
@@ -3,6 +3,65 @@
+    * replace `const chart` for `var chart` in advanced tutorial #1679
+* update LICENSE to 2023 #1683
+
+
```
Too many blank lines.

rht commented:
Why is the target branch v2.0? Shouldn't it be main?

tpike3 commented:
> Why is the target branch v2.0? Shouldn't it be main?

@jackiekazil told me to do to v2.0, if I remember correctly that then makes the release easier.

tpike3 commented:
Forgot to update the date

rht commented on HISTORY.rst:
```diff
@@ -3,6 +3,60 @@
+    * `position_agent()`: convert this to `place_agent`
+    * `neighbor_iter()`: convert this to `iter_neighborhood()`
+* batchrunner: remove deprecations #1627
+    * `batchrunner()` and `batchrunnerMP()`: convert these to `batch_run()`
```
Aren't those supposed to be `BatchRunner` and `BatchRunnerMP` classes, not functions?

rht commented on HISTORY.rst:
```diff
@@ -3,6 +3,60 @@
+    * `neighbor_iter()`: convert this to `iter_neighborhood()`
+* batchrunner: remove deprecations #1627
+    * `batchrunner()` and `batchrunnerMP()`: convert these to `batch_run()`
+* visualization: remove `UserSettableParameter`; easier visualization creation #1693
```
This is too terse. I will have no idea what to replace `UserSettableParameter` with.

tpike3 replied:
I put in some changes feel free to edit as well so we can get this out the door.

jackiekazil commented:
@tpike3, @rht I changed the target branch to Main -- rht is right. We never updated v2.0. 

#1726 was merged.

@tpike3 is everything tagged with the milestone? 
And can you do final updates on the history? 
I can deploy Thursday or Friday... but I have a slight preference for Saturday because I have more time if there are any issues. :-)

tpike3 commented:
> @tpike3, @rht I changed the target branch to Main -- rht is right. We never updated v2.0.
> 
> #1726 was merged.
> 
> @tpike3 is everything tagged with the milestone? And can you do final updates on the history? I can deploy Thursday or Friday... but I have a slight preference for Saturday because I have more time if there are any issues. :-)

Changed the date to Saturday the 15th

jackiekazil commented:
Is there a reason this was closed?

tpike3 commented:
> Is there a reason this was closed?

Apologies was rushing this was superseded by #1731 -- merge into main instead of v2.0
